### PR TITLE
(WIP) GH-1803

### PR DESCRIPTION
**Pull request by hestonhoffman**
Adding an "Experimental features" section for features that are large
enough to need documentation, but aren't neccessarily ready to be added
to the docs proper.

!no-release-note

### CODE REVIEW

File: documentation/experimental_features.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
+# Experimental features
+
+This page describes experimental features that are currently available in Bolt.
```
Comment (adreyer) [inline documentation/experimental_features.md:3]:
```suggestion
Most larger bolt features are released initially in an experimental or unstable state. This allows the bolt team to gather feedback from real users quickly while iterating on new functionality. Almost all experimental features are eventually stabilized in future bolt releases. While a feature is experimental it's API may change requiring the user to update their code or configuration. The bolt team attempts to make these changes painless and provide useful warnings around breaking behavior where possible. 
```

File: documentation/bolt_project_directories.md [hunk 2]
```diff
@@ -2,11 +2,27 @@
...
+
+**Note:** Projects take precedence over installed modules of the same name. 
 
 ## Types of project directories
 
-There are three types of project directories that you can use depending on how you're using Bolt.
+There are three types of project directories that you can use depending on how
+you're using Bolt.
+
+Avoid giving your project the same name as another module in your modulepath.
```
Comment (adreyer) [inline documentation/bolt_project_directories.md:25]: Does this belong in the previous section?

Comment (adreyer) [reply documentation/bolt_project_directories.md:25]: and does that section belong only in experimental?

Comment (hestonhoffman) [reply documentation/bolt_project_directories.md:25]: Yes. I'll remove/consilidate

**Pull request merged**